spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -611,27 +627,29 @@ def __regroup_per_tracts(self, vol_dct, tracts):
     def _measure_totLesion_distribution(self, im_lesion, atlas_data, im_vert, p_lst):
 
         sheet_name = 'ROI_occupied_by_lesion'
-        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict({'vert': [str(v) for v in self.vert_lst] + ['total % (all vert)']})
+        rows_with_total = {
+            **self.rows,
+            # numpy array index equivalent to [:, :, :]
+            f'total % (all {self.row_name})': (slice(None), slice(None), slice(None)),
+        }
+        self.distrib_matrix_dct[sheet_name] = pd.DataFrame.from_dict({'row': [str(r) for r in rows_with_total]})
 
         # initialized to 0 for each vertebral level and each PAM50 tract
         for tract_id in atlas_data:
-            self.distrib_matrix_dct[sheet_name]['PAM50_' + str(tract_id).zfill(2)] = [0] * len(self.vert_lst + ['total % (all vert)'])
+            self.distrib_matrix_dct[sheet_name]['PAM50_' + str(tract_id).zfill(2)] = [0] * len(rows_with_total)
 
-        for vert in self.vert_lst + ['total % (all vert)']:  # loop over the vertebral levels
-            if vert != 'total % (all vert)':
-                im_vert_cur = np.copy(im_vert)
-                im_vert_cur[np.where(im_vert_cur != vert)] = 0
-            else:
-                im_vert_cur = None
-            if im_vert_cur is None or np.count_nonzero(im_vert_cur * np.copy(im_lesion)):
+        im_vert_and_lesion = im_vert * im_lesion  # to check which vertebral levels have lesions
+        # loop over slices/vertlevels
+        for row, indices_to_keep in sct_progress_bar(rows_with_total.items(), unit=self.row_name,
+                                                     desc="  Computing ROI distribution for all lesions"):
+            if row.startswith('total') or np.count_nonzero(im_vert_and_lesion[indices_to_keep]):
```

Comment:
I think since `row` is an integer otherwise (slice/vert number), this needs to be `str(row)`?

Reply:
Fixed, thanks for catching this! I changed it back to an equality test, because that really captures the intent better.